site/en/guides/border-router/external-commissioning/cli.md

```diff
@@ -1,35 +1,80 @@
 
 External commissioning is supported by the OT Commissioner CLI, available
-on the [ot-commissioner GitHub repository](https://github.com/openthread/ot-commissioner). For
-instructions on how to build and install OT Commissioner, refer to [OT Commissioner Build and
-Configuration](https://openthread.io/guides/commissioner/build).
+on the [ot-commissioner GitHub repository](https://github.com/openthread/ot-commissioner).
+
+In this guide, you'll build and install OT Commissioner and commission a Joiner.
+
+## Set up OT Commissioner
+
+To use OT Commissioner CLI, first [Build OT Commissioner](../../commissioner/build.md).
 
 > Note: OT Commissioner must be running on the same host machine as OTBR.
 
+## Discover your network
+
+OTBR registers a `_meshcop._udp` service named `OpenThread_BorderRouter`. This
+service is configured in the [OTBR CMakeLists.txt file](https://github.com/openthread/ot-br-posix/blob/main/CMakeLists.txt#L36).
+
+To start OT Commissioner, you'll need to find the port number of your border
+agent service. There are several ways to do this, for example:
+
+*   Use `dns-sd` to browse for a service on `raspberrypi.local.:49155`:
+
+    ```
+    $ dns-sd -L "OpenThread_BorderRouter" _meshcop._udp local
+    ```
+
+    <pre>
+    Lookup OpenThread_BorderRouter._meshcop._udp.local
+    DATE: ---Tue 16 Nov 2021---
+    13:31:03.197  ...STARTING...
+    13:31:03.350  OpenThread_BorderRouter._meshcop._udp.local. can be reached at raspberrypi.local.:49155 (interface 3)
+    </pre>
+
+*   Use `avahi-utils`:
+
+    ```
+    $ avahi-browse -r _meshcop._udp
+    ```
+
+> Note: To support multiple Thread interfaces on a single host, the border agent
+service uses an ephemeral port by default. This means that your port number
+might change when you restart `otbr-agent` or your OTBR platform.
+
 ## Connect to the Border Router
 
-1.  Open the Non-CCM configuration file located at
-    `/usr/local/etc/commissioner/non-ccm-config.json` and change the `PSKc` to
-    `198886f519a8fd7c981fee95d72f4ba7`:
+1.  Open the Non-CCM configuration file in your installation directory:
+
+    ```
+    $ cd /usr/local/etc/commissioner/non-ccm-config.json
+    ```
+
+1.  Change the `PSKc` to `198886f519a8fd7c981fee95d72f4ba7`:
 
     ```
     "PSKc" : "198886f519a8fd7c981fee95d72f4ba7"
     ```
 
+    _Optional_. Configure logging. When you run `pi@raspberrypi: commissioner-cli`
+    from the command line, OT Commissioner creates a `commissioner.log` file in
+    the current working directory, for example `/home/pi/commissioner.log`. In
+    the `non-ccm-config.json` file, you can configure your `LogFile` path,
+    logging level, and other log settings.
+
 1.  Start the OT Commissioner CLI with the Non-CCM configuration:
 
     ```
     $ commissioner-cli /usr/local/etc/commissioner/non-ccm-config.json
     > 
     ```
 
-1.  Connect to OTBR:
+1.  Connect to OTBR, providing your OTBR port:
 
     ```
-    > start :: 49191
+    > start :: 49155
```

Comment:
Using the all-zero (`ANY`) IPv6 address is assuming that the OT Commissioner is on the same host of the Border Router. I think it would be better to use the IP addresses returned by the mDNS browsers, like `13:31:03.350` in the `dns-sd -L "OpenThread_BorderRouter" _meshcop._udp local` example output.